plugins/test/framework.cc

```diff
@@ -130,21 +192,43 @@ TestHttpContext::Result TestHttpContext::SendRequestHeaders(
   phase_logs_.clear();
   result_ = Result{.headers = std::move(headers)};
   phase_ = proxy_wasm::WasmHeaderMapType::RequestHeaders;
-  result_.status =
+  result_.header_status =
       onRequestHeaders(result_.headers.size(), /*end_of_stream=*/false);
   phase_ = proxy_wasm::WasmHeaderMapType(-1);  // ideally 0 would mean unset
   return std::move(result_);
 }
+
+TestHttpContext::Result TestHttpContext::SendRequestBody(
+    const std::string_view body) {
+  phase_logs_.clear();
+  result_ = Result{};
+  body_buffer_.set(body);
+  result_.body_status =
+      onRequestBody(body_buffer_.size(), /*end_of_stream=*/false);
+  result_.body = body_buffer_.get();
```

Comment:
This incurs a copy even if the body hasn't been changed.

To avoid a copy, I think you may need to change Result::body to be of type string_view, and also change Buffer::get() to return string_view as suggested in another comment.